src/test/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteArtifactsMatchingPatternsActionTest.java

```diff
@@ -0,0 +1,388 @@
+    }
+
+    @Rule
+    public TemporaryFolder tempFolder = TemporaryFolder.builder().assureDeletion().build();
+
+    @Before
+    public void setup() throws IOException {
+        action = new DeleteArtifactsMatchingPatternsAction();
+        mockRun = Mockito.mock(Run.class);
+        mockArtifactManager = Mockito.mock(ArtifactManager.class);
+        Mockito.when(mockRun.getArtifactManager()).thenReturn(mockArtifactManager);
+        VirtualFile root = VirtualFile.forFile(tempFolder.getRoot());
+        Mockito.when(mockArtifactManager.root()).thenReturn(root);
+
+        // Create the tempFolder and its contents
+        tempFolder.newFile("test.xml");
+        tempFolder.newFile("testLog.log");
+        tempFolder.newFile("testTxt.txt");
+        tempFolder.newFolder("testFolder");
+        tempFolder.newFile("testFolder/test1.xml");
+        tempFolder.newFile("testFolder/testLog1.log");
+        tempFolder.newFile("testFolder/testTxt1.txt");
+
+        // Create a temporary directory for code coverage testing
+        Path tempDir = Files.createTempDirectory("testDir");
+        archiveRootDirectory = tempDir.resolve("archive").toFile();
+        //Set up DeleteFileCallable instance with the archiveRootDirectory
+        deleteFileCallableInstance = new DeleteArtifactsMatchingPatternsAction.DeleteFileCallable(archiveRootDirectory);
+    }
+
+    private void assertTempFolderContains(final int expectedNumFiles) {
+        final int actualNumFiles = countFiles(tempFolder.getRoot());
+        final String assertMessage = String.format("Expected %d files, actual found %d files", expectedNumFiles,
+                actualNumFiles);
+        Assert.assertEquals(assertMessage, expectedNumFiles, actualNumFiles);
+    }
+
+    // Custom assertFileExists method
+    public void assertFileExists(final String filePath, final String message) {
+        final File file = new File(tempFolder.getRoot(), filePath);
+        Assert.assertTrue(message, file.exists());
+    }
+
+    public void assertFileExists(final String filePath) {
+        assertFileExists(filePath, null);
+    }
+
+    // Custom assertFileNotExists method
+    public void assertFileNotExists(final String filePath, final String message) {
+        final File file = new File(tempFolder.getRoot(), filePath);
+        Assert.assertFalse(message, file.exists());
+    }
+
+    public void assertFileNotExists(final String filePath) {
+        assertFileNotExists(filePath, null);
+    }
+
+    // Count the number of files in the directory
+    public static int countFiles(final File directory) {
+        int fileCount = 0;
+        File[] files = directory.listFiles();
+        assert files != null;
+        for (File file : files) {
+            if (file.isFile() && file.exists()) {
+                fileCount++;
+            } else if (file.isDirectory()) {
+                fileCount += countFiles(file);
+            }
+        }
+        return fileCount;
+    }
```

Comment:
did you check this https://www.baeldung.com/java-list-directory-files ?

Reply:
Are you suggesting an alternative method `listFilesUsingJavaIO` to get the names of files in a directory? This method uses Java Streams to filter out directories and gather file names into a set. However, it has a slightly different goal compared to the `countFiles` method, which is designed to count files rather than list their names.